pkg/controllers/nodeclass/controller.go

```diff
@@ -87,7 +89,7 @@ func NewController(
 			&Subnet{subnetProvider: subnetProvider},
 			&SecurityGroup{securityGroupProvider: securityGroupProvider},
 			&InstanceProfile{instanceProfileProvider: instanceProfileProvider},
-			&Validation{ec2api: ec2api, amiProvider: amiProvider},
+			NewValidationReconciler(ec2api, amiProvider, validationCache),
```

Comment:
Why are we changing this? If it's a preference, I'd prefer to see all of these reconcilers updated in one go.

Reply:
We were doing them ad-hoc, it's already been done for ODCR and AMIs.

Reply:
I can knock the rest out here though, shouldn't take long